components/user-mgt/org.wso2.carbon.identity.user.profile/src/main/java/org/wso2/carbon/identity/user/profile/mgt/UserProfileAdmin.java

```diff
@@ -850,60 +846,87 @@ private boolean isAdminProfileSpoof(String username) throws UserStoreException {
-        }
+    private FederatedAssociationManager getFederatedAssociationManager() throws UserProfileException{
 
-        try {
-            UserStoreManager userStoreManager = getUserRealm().getUserStoreManager();
-            if (!userStoreManager.isExistingUser(username)) {
-                throw new UserProfileException("UserNotFound: User: " + username + " does not exist in tenant: " +
-                        getTenantDomain());
+        FederatedAssociationManager federatedAssociationManager = IdentityUserProfileServiceDataHolder.getInstance()
+                .getFederatedAssociationManager();
+        if (federatedAssociationManager == null) {
+            if (log.isDebugEnabled()) {
+                log.debug("FederatedAssociationManager is not available in the OSGi framework");
             }
-        } catch (UserStoreException e) {
-            String msg = "Error occurred while verifying user: " + username + " exists in tenant: " + getTenantDomain();
-            log.error(msg, e);
-            throw new UserProfileException(msg, e);
+            String msg = "Error while working with federated associations";
+            log.error(msg);
+            throw new UserProfileException(msg);
         }
+        return federatedAssociationManager;
     }
 
-    private void audit(String action, String target, String data, String result) {
+    private String getFullyQualifiedUserNameFromDomainAwareUserName(String domainAwareUserName) {
 
-        audit_log.info(String.format(AUDIT_MESSAGE, getUsername() + UserCoreConstants.TENANT_DOMAIN_COMBINER +
-                getTenantDomain(), action, target, data, result));
+        User user = new User();
+        user.setTenantDomain(CarbonContext.getThreadLocalCarbonContext().getTenantDomain());
+        user.setUserStoreDomain(UserCoreUtil.extractDomainFromName(domainAwareUserName));
+        user.setUserName(MultitenantUtils.getTenantAwareUsername(UserCoreUtil.removeDomainFromName(domainAwareUserName)));
+        return user.toFullQualifiedUsername();
     }
 
-    private String getAuditData(String username, String idpID, String federatedUserID) {
+    private boolean isFederatedAssociationDoesNotExistsError(FederatedAssociationManagerException e) {
 
-        return "\"" + "Username" + "\" : \"" + username + "\", " + "\"" + "IdP" + "\" : \"" + idpID + "\", " + "\"" +
-                "FederatedID" + "\" : \"" + federatedUserID + "\"";
+        return e.getErrorCode().contains(String.valueOf(INVALID_FEDERATED_ASSOCIATION.getCode()));
     }
 
+    private void validateFederatedAssociationParameters(String username, String idpID, String associatedID,
+                                                        String auditData) throws UserProfileException {
+
+        if (StringUtils.isEmpty(idpID) || StringUtils.isEmpty(associatedID)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Required parameters, idpId and the associatedId are empty.");
```

Comment:
idpId or the associatedId are empty.

Reply:
Fixed in 16a9f7f8559dca533109b50dd6036a753511b309